### PR TITLE
Upgrade to Substrate 2.0

### PR DESCRIPTION
**Pull request by AurevoirXavier**
https://github.com/paritytech/substrate.git?rev=c2fccb36ffacd118fc3502aa93453580a07dc402#c2fccb36

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
**bug in this version**
- https://github.com/paritytech/substrate/issues/4220

**Comment by AurevoirXavier** [comment]
**TODO**:
1. core/* upgrade
2. chainrelay upgrade (depends on 1)
3. staking
	- new slash in staking
	- use nicks #176

File: Cargo.toml [hunk 1]
```diff
@@ -1,27 +1,18 @@
...
-    "srml/eth-backing",
-    "srml/im-online",
-    "srml/kton",
-    "srml/staking",
-    "srml/support",
+	"bin/node/cli",
+	"bin/node/executor",
+    "bin/node/primitives",
+    "bin/node/rpc-client",
+    "bin/node/rpc",
```
Comment (yanganto) [inline Cargo.toml]: There may be a problem mixing tabs and space

File: README.adoc [hunk 2]
```diff
@@ -339,20 +339,33 @@ darwinia-balances, darwinia-eth-backing, darwinia-eth-relay,
...
 === Environment
 
+<<<<<<< HEAD
+If you are using Ubuntu, 
+you may use the `.maintain/bootstrap.sh` to set up your develop environment.
+In this script, the nightly `Rust`, `cargo`, `rustfmt` will be installed,
+=======
 The `scripts/bootstrap.sh` help set up your develop environment.
 The nightly `Rust`, `cargo`, `rustfmt` will be installed,
+>>>>>>> upstream/develop
```
Comment (yanganto) [inline README.adoc]: There are some conflicts we need to solve here.

File: Cargo.toml [hunk 3]
```diff
@@ -1,27 +1,18 @@
...
-
-    "srml/balances",
-    "srml/eth-relay",
-    "srml/eth-backing",
-    "srml/im-online",
-    "srml/kton",
-    "srml/staking",
-    "srml/support",
+	"bin/node/cli",
+	"bin/node/executor",
```
Comment (yanganto) [inline Cargo.toml]: There may be a problem mixing tabs and space

File: README.adoc [hunk 4]
```diff
@@ -339,20 +339,33 @@ darwinia-balances, darwinia-eth-backing, darwinia-eth-relay,
...
 if you want to do it manually, use the `--fast` option to skip.
 We will appreciate your contribution.
 
+<<<<<<< HEAD
+If you are using different environment, you may copy the git hooks mannually.
+```
+$ cp .maintain/hooks/* .git/hooks
+```
+=======
+>>>>>>> upstream/develop
```
Comment (yanganto) [inline README.adoc]: The git merging marks should be removed.

File: bin/node/cli/src/browser.rs [hunk 5]
```diff
@@ -56,14 +55,9 @@ fn start_inner(wasm_ext: wasm_ext::ffi::Transport) -> Result<Client, Box<dyn std
...
 	};
 
-	info!("Darwinia browser node");
-	info!("Version: {}", config.full_version());
-	info!("  _____                      _       _       ");
-	info!(" |  __ \\                    (_)     (_)      ");
-	info!(" | |  | | __ _ _ ____      ___ _ __  _  __ _ ");
-	info!(" | |  | |/ _` | '__\\ \\ /\\ / / | '_ \\| |/ _` |");
-	info!(" | |__| | (_| | |   \\ V  V /| | | | | | (_| |");
-	info!(" |_____/ \\__,_|_|    \\_/\\_/ |_|_| |_|_|\\__,_|");
```
Comment (yanganto) [inline bin/node/cli/src/browser.rs]: The logo is good, should we remove this?

File: bin/node/cli/bin/main.rs [hunk 6]
```diff
@@ -18,45 +18,40 @@
...
+fn main() -> Result<(), sc_cli::error::Error> {
 	let version = VersionInfo {
-		name: "Darwinia IceFrog Node",
+		name: "Substrate Node",
 		commit: env!("VERGEN_SHA_SHORT"),
 		version: env!("CARGO_PKG_VERSION"),
-		executable_name: "darwinia",
-		author: "Darwinia Network <hello@darwinia.network>",
-		description: "Darwinia poc-1 node",
-		support_url: "https://github.com/darwinia-network/darwinia/issues/new",
```
Comment (yanganto) [inline bin/node/cli/bin/main.rs]: Could we keep the correct links let others can report bugs to our GitHub?

**Comment by yanganto** [comment]
@AurevoirXavier Thanks for upgrading.  There are many things change between Substrate 1.0 and 2.0.     
 Really hard work.  This PR can not be reviewed by me, because I do not have good enough knowledge with Substrate now.  @HackFisher May you help to review this PR.  Thanks.

File: CONTRIBUTING.adoc [hunk 7]
```diff
@@ -1,62 +0,0 @@
-= Contributing
```
Comment (yanganto) [inline CONTRIBUTING.adoc]: May we keep this file?

File: bin/node/rpc-client/Cargo.toml [hunk 8]
```diff
@@ -10,7 +10,6 @@ futures = "0.1.29"
 hyper = "0.12.35"
 jsonrpc-core-client = { version = "14.0.3", features = ["http", "ws"] }
 log = "0.4.8"
-
-substrate-rpc = { git = "https://github.com/darwinia-network/substrate.git", branch = "darwinia-develop", version = "2.0.0" }
+sc-rpc = { version = "2.0.0", git = "https://github.com/paritytech/substrate.git", rev = "c2fccb36ffacd118fc3502aa93453580a07dc402" }
```
Comment (HackFisher) [inline bin/node/rpc-client/Cargo.toml]: Better to use darwinia's repo as dependency version control.

**Comment by yanganto** [comment]
It may be better to rename the folder `frame` to `pallet`.
Because all this modules are imported with `pallet-` prefix.
```
pallet-kton = { package = "darwinia-kton", default-features = false, path = "../../../frame/balances/kton" }
pallet-ring = { package = "darwinia-ring", default-features = false, path = "../../../frame/balances/ring" }
pallet-staking = { package = "darwinia-staking", default-features = false, features = ["migrate"], path = "../../../frame/staking" }
```

**Comment by AurevoirXavier** [comment]
> It may be better to rename the folder `frame` to `pallet`.
> Because all this modules are imported with `pallet-` prefix.
> 
> ```
> pallet-kton = { package = "darwinia-kton", default-features = false, path = "../../../frame/balances/kton" }
> pallet-ring = { package = "darwinia-ring", default-features = false, path = "../../../frame/balances/ring" }
> pallet-staking = { package = "darwinia-staking", default-features = false, features = ["migrate"], path = "../../../frame/staking" }
> ```

Better to follow substrate style

**Comment by AurevoirXavier** [comment]
> Build checks failed.

Could you paste your log here

**Comment by HackFisher** [comment]
> > Build checks failed.
> 
> Could you paste your log here

https://travis-ci.org/darwinia-network/darwinia/jobs/646853550?utm_medium=notification&utm_source=github_status

**Pull request merged**